stubs/docker/docker/models/images.pyi

```diff
@@ -1,10 +1,21 @@
 from collections.abc import Iterator
-from typing import Any, Literal, overload
+from io import StringIO
+from typing import IO, Any, Literal, TypedDict, overload, type_check_only
 from typing_extensions import TypeAlias
 
 from .resource import Collection, Model
 
 _ImageList: TypeAlias = list[Image]  # To resolve conflicts with a method called "list"
+# Type alias for JSON, explained at:
+# https://github.com/python/typing/issues/182#issuecomment-1320974824.
+_JSON: TypeAlias = dict[str, _JSON] | list[_JSON] | str | int | float | bool | None
```

Comment:
I'm open to moving this to a common place (as it is duplicated from `docker.utils.json_stream` - but if so, I'd like a suggestion on where that should go / what it should be named, please.

Reply:
We could add a typeshed-only module like `docker._types`.

Reply:
I will try that and then submit this PR. Thank you!

Reply:
@srittau I have done this and marked this PR as ready for review.